SliceTracker/SliceTracker.py

```diff
@@ -2880,16 +2880,24 @@ def applyRegistration(self, progressCallback=None):
 
     coverProstateRegResult = self.registrationResults.getMostRecentApprovedCoverProstateRegistration()
     lastRigidTfm = self.registrationResults.getLastApprovedRigidTransformation()
+    lastApprovedTfm = self.registrationResults.getMostRecentApprovedTransform()
+    initialTransform = lastApprovedTfm if lastApprovedTfm else lastRigidTfm
 
     self.generateNameAndCreateRegistrationResult(self.currentIntraopVolume)
     parameterNode = slicer.vtkMRMLScriptedModuleNode()
     parameterNode.SetAttribute('FixedImageNodeID', self.currentIntraopVolume.GetID())
-    parameterNode.SetAttribute('FixedLabelNodeID', coverProstateRegResult.fixedLabel.GetID())
+
+    fixedLabel = self.volumesLogic.CreateAndAddLabelVolume(slicer.mrmlScene, self.currentIntraopVolume,
+                                                           self.currentIntraopVolume.GetName() + '-label')
+
+    self.runBRAINSResample(inputVolume=coverProstateRegResult.fixedLabel, referenceVolume=self.currentIntraopVolume,
+                           outputVolume=fixedLabel, warpTransform=initialTransform)
```

Comment:
Correction: I misunderstood, `coverProstateRegResult` is the result of registering preop T2 to cover prostate, so `coverProstateRegResult.fixedLabel` is the segmentation of the prostate in the cover prostate scan, as expected.